### PR TITLE
14964 mobile study options

### PR DESCRIPTION
**Pull request by jdrichards50**
Fixed a number of issues with the study options page. Mainly stabilizing the behavior of displaying and persisting study option selections.

### CODE REVIEW

File: app/src/main/java/com/chartiq/chartiqsample/studies/StudiesActivity.java [hunk 1]
```diff
@@ -165,10 +160,21 @@ public void onClick(View v) {
...
-                activeStudies.addAll(lastSelection);
-                studiesAdapter.setActiveStudiesList(activeStudies);
-                studiesAdapter.getAvailableStudies().removeAll(lastSelection);
-                clearSelection();
+                //activeStudies.addAll(lastSelection);
+                //studiesAdapter.setActiveStudiesList(activeStudies);
+                //studiesAdapter.getAvailableStudies().removeAll(lastSelection);
+                //clearSelection();
+
+                chartIQ.getActiveStudies().than(new Promise.Callback<Study[]>() {
```
Comment (rbruckheimer) [inline app/src/main/java/com/chartiq/chartiqsample/studies/StudiesActivity.java]: Is `than` proper syntax or a spelling error?

File: app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java [hunk 2]
```diff
@@ -260,30 +351,16 @@ public void onClick(View v) {
...
-                            Integer.parseInt(rgbColors[1].trim()), Integer.parseInt(rgbColors[2].trim()));
-
-                    color.setBackgroundColor(parsedColor);
-                } else {
-                    color.setBackgroundColor(Color.parseColor(parameter.color));
-                }
-            }catch(Exception e) {
-                e.printStackTrace();
+        } else if(parameter.color != null){
+            color.setBackgroundColor(Color.parseColor(parameter.color));
```
Comment (rbruckheimer) [inline app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java]: I'm should assume that `Color.parseColor` will handle the rgb values that the removed code once did?

**Comment by jdrichards50** [comment]
> Since there is no way I can tell if you've caught every case where parameters need to be checked, I'm taking it on faith that you've tested this and it works. Does the ios app need to be changed too?

Tested all three study options and everything works. Doesn't mean there isn't an edge case lurking. 

iOS will need some tweaking as well, that'll be covered in 15379.

File: app/src/main/java/com/chartiq/chartiqsample/studies/StudiesActivity.java (hunk 1 above)
Comment (jdrichards50) [reply app/src/main/java/com/chartiq/chartiqsample/studies/StudiesActivity.java]: Promise is a class that we control, so a typo from when it was first written. There are only a few instances where we use a Promise so it'll be simple enough to change.

File: app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java (hunk 2 above)
Comment (jdrichards50) [reply app/src/main/java/com/chartiq/chartiqsample/studies/StudyOptionsActivity.java]: This is odd... I branched off master but the changes for rgb committed last month aren't there. Something is wrong.

**Pull request closed without merging**